extensions/mypy_extensions.py

```diff
@@ -131,3 +131,13 @@ def KwArg(type=Any):
 
 # Return type that indicates a function does not return
 class NoReturn: pass
+
+
+def declared_type(t):
+    """Declare the type of a declaration.
+
+    This is useful for declaring a more specific type for a decorated class or
+    function definition than the decorator provides as a return value.
```

Comment:
I wonder if there is a context where it's useful to use this to decorate a class?